tools/fuzzer/constraints.h

```diff
@@ -38,6 +38,7 @@ class TypeConstraints;
 
 enum class VoidPointerConstraint : bool { Deny, Allow };
 enum class ExprCategory : bool { LvalueOrRvalue, Lvalue };
+enum class PointerValidity : bool { ValidOrInvalid, Valid };
```

Comment:
It seemed to me that we already tend to use enums rather than booleans, I guessed for better readability and to avoid mixing arguments up in function calls, e.g. `ExprConstraint(ExprCategory, PointerValidity)` vs `ExprConstraint(bool /* must_be_lvalue */, bool /* must_be_valid_pointer */)`.
Anyway, it can be a boolean and I don't have anything against converting this to boolean.